lib/thredded/per_notifier_pref.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+module Thredded
+  # in this class we dynamically set up an accessor for all the Thredded.notifiers (by their key)
+  # as we don't know in advance what the keys are,
+  # we try to not pollute the namespace
+  # nonetheless some keys would be bad, so we sanity check those on setup
+  class PerNotifierPref
```

Comment:
Perhaps it might be better to make 2 join tables of (notifier_key, value) and one of (messageboard_id, notifier_key, value)?

The benefits are:
1. No custom de/serialization code needed for the database.
2. No custom parsing code needed for the URL params.
3. Queries such as "users who opted out/in of pushover notifications [in the last month] [etc]" are possible.
4. The UI code becomes simpler, as the form object can provide the nested form objects:

```ruby
def notifications_for_followed_topics
  Thredded.notifications.map do |notification_key|
    @user.notifications_for_followed_topics.detect { |n| n.key == notification_key } ||
      @user.notifications_for_followed_topics.build(notification_key: notification_key)
  end
end
```